### ISSUE TITLE
Generated MP4 not working with MediaPlayer

### ISSUE THREAD

**Issue by zadokov** [lllyasviel/FramePack#15]
Getting a black screen when trying to view with Media Player, but video works in Photoshop or clipchamp.
If I render the video it is then visible correctly.
What video encoder should be used/installed?

**Comment by tristan88888** [comment]
Temporary patch to line 280 of diffusers_helper/utils.py

try changing the video_codec and crf parameter values...

torchvision.io.write_video(output_filename, x, fps=fps, video_codec='libx264', options={'crf': '18'})

**Comment by tristan88888** [comment]
Alternatively if your system has FFMPEG installed, can explore using that to generate the mp4 file (slightly faster on my machine)

import os
import torch
import einops
import subprocess

def save_bcthw_as_mp4(x, output_filename, fps=10):
    b, c, t, h, w = x.shape

    # Determine grid layout
    per_row = b
    for p in [6, 5, 4, 3, 2]:
        if b % p == 0:
            per_row = p
            break

    # Normalize and convert to uint8 RGB
    x = torch.clamp(x.float(), -1., 1.) * 127.5 + 127.5
    x = x.detach().cpu().to(torch.uint8)
    x = einops.rearrange(x, '(m n) c t h w -> t (m h) (n w) c', n=per_row)

    t, H, W, C = x.shape
    assert C == 3, "Only RGB videos are supported"

    os.makedirs(os.path.dirname(os.path.abspath(output_filename)), exist_ok=True)

    # Fully compatible ffmpeg command
    cmd = [
        'ffmpeg',
        '-y',  # Overwrite output
        '-f', 'rawvideo',
        '-vcodec', 'rawvideo',
        '-pix_fmt', 'rgb24',
        '-s', f'{W}x{H}',
        '-r', str(fps),
        '-i', '-',  # Input from stdin
        '-vf', 'format=yuv420p',
        '-c:v', 'h264_nvenc',
        '-preset', 'p4',
        '-qp', '18',
        '-movflags', '+faststart',
        '-f', 'mp4',
        output_filename
    ]

    # Use subprocess to stream frames to ffmpeg
    proc = subprocess.Popen(cmd, stdin=subprocess.PIPE)
    try:
        proc.stdin.write(x.numpy().tobytes())
        proc.stdin.close()
        proc.wait()
    except Exception as e:
        proc.kill()
        raise RuntimeError(f"ffmpeg failed: {e}")

    return x

**Comment by Farrenzo** [comment]
Thanks for your fix with the FFMPEG, it really does run faster.

**Comment by zadokov** [comment]
This is fixed now in the latest version! Thanks @lllyasviel !!!
